File: epi_judge_java/epi/PowerSet.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.EpiTestComparator;
import epi.test_framework.GenericTest;
import epi.test_framework.LexicographicalListComparator;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class PowerSet {

    @EpiTest(testDataFile = "power_set.tsv")
    public static List<List<Integer>> generatePowerSet(List<Integer> inputSet) {
        List<List<Integer>> result = new ArrayList<>();
        int n = inputSet.size();

        for (int i = 0; i < (1 << n); i++) {
            List<Integer> next = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if ((i & (1 << j)) != 0) {
                    next.add(inputSet.get(j));
                }
            }

            result.add(next);
        }

        return result;
    }

    @EpiTestComparator
    public static boolean comp(List<List<Integer>> expected,
                               List<List<Integer>> result) {
        if (result == null) {
            return false;
        }
        for (List<Integer> l : expected) {
            Collections.sort(l);
        }
        expected.sort(new LexicographicalListComparator<>());
        for (List<Integer> l : result) {
            Collections.sort(l);
        }
        result.sort(new LexicographicalListComparator<>());
        return expected.equals(result);
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "PowerSet.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/power_set.cc
#include <vector>

#include "test_framework/generic_test.h"
using std::vector;

vector<vector<int> > GeneratePowerSet(const vector<int> &input_set) {
    int n = input_set.size();
    vector<vector<int>> ans;
    for (int mask = 0; mask < (1 << n); mask++) {
        vector<int> new_set;
        for (int i = 0; i < n; i++) {
            if ((mask >> i) & 1) {
                new_set.push_back(input_set[i]);
            }
        }
        ans.push_back(new_set);
    }
    return ans;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"input_set"};
    return GenericTestMain(args, "power_set.cc", "power_set.tsv",
                           &GeneratePowerSet, UnorderedComparator{}, param_names);
}
